SMTInterpol/src/de/uni_freiburg/informatik/ultimate/smtinterpol/convert/Clausifier.java

```diff
@@ -486,7 +487,7 @@ public void perform() {
 			if (mSubRewrites.length == 1) {
 				rewrite = mTracker.transitivity(mRewriteProof, mSubRewrites[0]);
 			} else {
-				rewrite = mTracker.congruence(mRewriteProof, mSubRewrites);
+				rewrite = mTracker.orMonotony(mRewriteProof, mSubRewrites);
```

Comment:
It may be better to separate the transitivity from the `orMonotony` rule.  I don't really like this for congruence, but there it avoids to specify the function symbol.  Here the function symbol is only `or`.

However, changing it is only possible if mSubRewrites is guaranteed to contains every argument of the or, even those that aren't rewritten (i.e. it needs to contain `@refl` arguments).  I haven't checked that this is the case here.

Reply:
The theoretical problem with this is that orMonotony is a left associative function.  If
 ```
    (orMonotony 'proof for a1=b1'  'proof for a2=b2') is a proof for (or a1 a2) = (or b1 b2)
```
then
```
   (orMonotony 'proof for a1=b1'  'proof for a2=b2' 'proof for a3=b3') would be a proof for (or (or a1 a2) a3) = (or (or b1 b2) b3)
```
but it should be a proof for the different (but equivalent) formula: `(or a1 a2 a3)=(or b1 b2 b3)`.  
We would really want orMonotony to be a non-associative function that takes arbitrary number of arguments, but that isn't possible to define.

So I keep it the way it is now.